build-tools/packages/build-cli/src/commands/generate/entrypoints.ts

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "node:fs/promises";
+import path from "node:path";
+
+import { Flags } from "@oclif/core";
+import type { ExportSpecifierStructure } from "ts-morph";
+import { ModuleKind, Project, ScriptKind } from "ts-morph";
+
+import { BaseCommand } from "../../base.js";
+// eslint-disable-next-line import/no-internal-modules
+import { ApiLevel } from "../../library/apiLevel.js";
+// eslint-disable-next-line import/no-internal-modules
+import { getApiExports } from "../../library/typescriptApi.js";
```

Comment:
The typical pattern in build-cli is to either export the API you want to share in library/index.ts, or add an index.js file to the folder in library and export the supported stuff from there. Either is fine, but reaching in is not.